src/AdsClient/AmsSocketConnection.cs

```diff
@@ -6,22 +6,24 @@
 
 namespace Viscon.Communication.Ads;
 
-internal class AmsSocketConnection
+internal sealed class AmsSocketConnection
 {
     private const int ReceiveTaskTimeout = 3000;
 
     private readonly Socket socket;
     private readonly IIncomingMessageHandler messageHandler;
     private readonly Task receiveTask;
+    private readonly SocketAwaitable socketAwaitable = new SocketAwaitable();
+    private readonly byte[] headerBuffer = new byte[AmsHeaderHelper.AmsTcpHeaderSize];
 
     public AmsSocketConnection(Socket socket, IIncomingMessageHandler messageHandler)
     {
         this.socket = socket;
         this.messageHandler = messageHandler;
-        receiveTask = Task.Run(ReceiveLoop);
+        receiveTask = ReceiveLoop();
     }
 
-    private bool closed;
+    private volatile bool closed;
```

Comment:
To be on the safe side, this should be `volatile` as in the receive-loop it's checked (read) and at that point different threads are in play. 

For x86-machines it's unlikely to be a real problem, but for good code hygiene and for platforms with a weak memory model it's a good addition. And no harm to x86-machines.